tools/cli/domainCommands.go

```diff
@@ -854,6 +911,107 @@ func (d *domainCLIImpl) migrationDomainWorkFlowCheck(c *cli.Context) DomainMigra
 	}
 }
 
+func (d *domainCLIImpl) searchAttributesChecker(c *cli.Context) DomainMigrationRow {
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	// getting user provided search attributes
+	searchAttributes := c.StringSlice(FlagSearchAttribute)
+	if len(searchAttributes) == 0 {
+		return DomainMigrationRow{
+			ValidationCheck:  "Search Attributes Check",
+			ValidationResult: true,
+		}
+	}
+
+	// Parse the provided search attributes into a map[string]IndexValueType
+	requiredAttributes := make(map[string]types.IndexedValueType)
+	for _, attr := range searchAttributes {
+		parts := strings.SplitN(attr, ":", 2)
+		if len(parts) != 2 {
+			ErrorAndExit(fmt.Sprintf("Invalid search attribute format: %s", attr), nil)
+		}
+		key, valueType := parts[0], parts[1]
+		ivt, err := parseIndexedValueType(valueType)
+		if err != nil {
+			ErrorAndExit(fmt.Sprintf("Invalid search attribute type for %s: %s", key, valueType), err)
+		}
+		requiredAttributes[key] = ivt
+	}
+
+	d.frontendClient = initializeFrontendClient(c)
+
+	// getting search attributes for current domain
+	currentSearchAttributes, err := d.frontendClient.GetSearchAttributes(ctx)
+	if err != nil {
+		ErrorAndExit("Unable to get search attributes for current domain.", err)
+	}
+
+	d.destinationClient = newClientFactory(func(c *cli.Context) string {
+		return c.String(FlagDestinationAddress)
+	}).ServerFrontendClient(c)
+
+	// getting search attributes for new domain
+	destinationSearchAttributes, err := d.destinationClient.GetSearchAttributes(ctx)
+	if err != nil {
+		ErrorAndExit("Unable to get search attributes for new domain.", err)
+	}
+
+	currentSearchAttrs := currentSearchAttributes.Keys
+	destinationSearchAttrs := destinationSearchAttributes.Keys
+
+	// checking to see if search attributes exist
+	missingInCurrent := findMissingAttributes(requiredAttributes, currentSearchAttrs)
+	missingInNew := findMissingAttributes(requiredAttributes, destinationSearchAttrs)
+
+	validationResult := len(missingInCurrent) == 0 && len(missingInNew) == 0
+
+	validationRow := DomainMigrationRow{
+		ValidationCheck:  "Search Attributes Check",
+		ValidationResult: validationResult,
+		ValidationDetails: ValidationDetails{
+			MissingCurrSearchAttributes: missingInCurrent,
+			MissingNewSearchAttributes:  missingInNew,
+		},
+	}
+
+	return validationRow
+}
+
+// helper to parse types.IndexedValueType from string
+func parseIndexedValueType(valueType string) (types.IndexedValueType, error) {
```

Comment:
use Unmarshal access method to avoid duplication logic. see example here https://sourcegraph.com/github.com/shijiesheng/cadence/-/blob/common/util.go?L805#L807:42